x/sponsorship/keeper/hooks.go

```diff
@@ -0,0 +1,238 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+
+	"cosmossdk.io/collections"
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/dymensionxyz/dymension/v3/x/sponsorship/types"
+)
+
+var _ stakingtypes.StakingHooks = Hooks{}
+
+// Hooks wrapper struct for slashing keeper
+type Hooks struct {
+	k Keeper
+}
+
+func (k Keeper) Hooks() Hooks {
+	return Hooks{k: k}
+}
+
+type processHookResult struct {
+	distribution     types.Distribution
+	votePruned       bool
+	vpDiff, newTotal math.Int
+}
+
+func (h Hooks) AfterDelegationModified(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) error {
+	err := h.afterDelegationModified(ctx, delAddr, valAddr)
+	if err != nil {
+		return fmt.Errorf("sponsorship: AfterDelegationModified: delegator '%s', validator '%s': %w", delAddr, valAddr, err)
+	}
+	return nil
+}
+
+func (h Hooks) afterDelegationModified(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) error {
+	voted, err := h.k.Voted(ctx, delAddr)
+	if err != nil {
+		return fmt.Errorf("cannot verify if the delegator voted: %w", err)
+	}
+
+	// Skip the vote if the delegator doesn't have a vote
+	if !voted {
+		return nil
+	}
+
+	v, found := h.k.stakingKeeper.GetValidator(ctx, valAddr)
+	if !found {
+		return fmt.Errorf("validator not found")
+	}
+
+	d, found := h.k.stakingKeeper.GetDelegation(ctx, delAddr, valAddr)
+	if !found {
+		return fmt.Errorf("delegation not found")
+	}
+
+	// Calculate a staking voting power
+	stakingVP := v.TokensFromShares(d.GetShares()).Ceil().TruncateInt()
+
+	// Get the current voting power saved in x/sponsorship. If the VP is not found, then we yet don't
+	// have a relevant record. This is a valid case when the VP is zero.
```

Comment:
not sure I get this. when we vote, we iterate over all delegator's validators (i.e where he put some delegation), and save for each delegator it's validator power:

`err = k.SaveDelegatorValidatorPower(ctx, voter, valPower.ValAddr, valPower.Power)`

assuming we've got the vote (which means the delegator voted for sponshership), why would it's delegator validator power will be zero or not found?